src/stream-tcp.c

```diff
@@ -927,6 +928,9 @@ static int StreamTcpPacketStateNone(ThreadVars *tv, Packet *p,
         if (!stream_config.midstream && stream_config.async_oneside == FALSE)
             return 0;
 
+        /* Drop reason will only be used if midstream policy is set to fail closed */
+        ExceptionPolicyApply(p, stream_config.midstream_policy, PKT_DROP_REASON_FLOW_DROP);
```

Comment:
I was wondering if we would want to create a specific drop reason for this policy

Reply:
yeah probably. Something like `PKT_DROP_REASON_STREAM_MIDSTREAM`?

Reply:
works for me!

Reply:
If we do add this reason, do we want to propagate that to all packet drops of a given flow, or is logging the first one with that enough?

Reply:
just the first one